File: libschema/src/symbol/argument.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Argument {
    /// Type of the argument
    pub r#type: String,

    // Name of the argument
    pub name: String,

    /// Raw declaration of the argument
    pub decl: String,

    /// Default value if any
    pub default: Option<String>,
}

File: typings/src/symbols/argument.ts
export interface Argument {
    /**
     * @brief Type of the argument
     * @readonly
     */
    readonly type: string,

    /**
     * @brief Name of the argument
     * @readonly
     */
    readonly name: string,

    /**
     * @brief Raw declaration of the argument
     * @readonly
     */
    readonly decl: string,

    /**
     * @brief Default value if any
     * @readonly
     */
    readonly default: string | null,
}
